crates/ruff_linter/src/rules/ruff/rules/invalid_formatter_suppression_comment.rs

```diff
@@ -0,0 +1,278 @@
+            // Ensure the comment is not a trailing comment
+            if !comment.line_position.is_own_line() {
+                return Err(IgnoredReason::FmtOnCannotBeTrailing);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn sort(&mut self) {
+        self.captured.sort_by_key(|(t, _)| t.start());
+    }
+
+    fn ignored_comments(&self) -> impl Iterator<Item = (TextRange, IgnoredReason)> + '_ {
+        self.captured.iter().map(|(r, i)| (*r, *i))
+    }
+}
+
+impl<'src, 'loc> CaptureSuppressionComment<'src> for UselessSuppressionComments<'src, 'loc> {
+    fn capture(&mut self, comment: SuppressionCommentData<'src>) {
+        match self.check_suppression_comment(&comment) {
+            Ok(()) => {}
+            Err(reason) => {
+                self.captured.push((comment.range, reason));
+            }
+        }
+    }
+}
+
+/// Returns `true` if `statement` is the first statement in an alternate `body` (e.g. the else of an if statement)
+fn is_first_statement_in_alternate_body(statement: AnyNodeRef, has_body: AnyNodeRef) -> bool {
+    match has_body {
+        AnyNodeRef::StmtFor(ast::StmtFor { orelse, .. })
+        | AnyNodeRef::StmtWhile(ast::StmtWhile { orelse, .. }) => {
+            are_same_optional(statement, orelse.first())
+        }
+
+        AnyNodeRef::StmtTry(ast::StmtTry {
+            handlers,
+            orelse,
+            finalbody,
+            ..
+        }) => {
+            are_same_optional(statement, handlers.first())
+                || are_same_optional(statement, orelse.first())
+                || are_same_optional(statement, finalbody.first())
+        }
+
+        AnyNodeRef::StmtIf(ast::StmtIf {
+            elif_else_clauses, ..
+        }) => are_same_optional(statement, elif_else_clauses.first()),
+        _ => false,
+    }
+}
+
+/// Returns `true` if `right` is `Some` and `left` and `right` are referentially equal.
+fn are_same_optional<'a, T>(left: AnyNodeRef, right: Option<T>) -> bool
+where
+    T: Into<AnyNodeRef<'a>>,
+{
+    right.is_some_and(|right| left.ptr_eq(right.into()))
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum IgnoredReason {
+    InsideExpression,
+    BetweenDecorators,
+    SkipHasToBeTrailing,
+    FmtOnCannotBeTrailing,
+    FmtOffAboveBlock,
+}
+
+impl Display for IgnoredReason {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::InsideExpression => write!(f, "it cannot be inside an expression"),
+            Self::BetweenDecorators => {
+                write!(f, "it cannot be between decorators")
+            }
+            Self::SkipHasToBeTrailing => {
+                write!(f, "it cannot be on its own line")
+            }
+            Self::FmtOnCannotBeTrailing => {
+                write!(f, "it cannot be at the end of a line")
+            }
+            Self::FmtOffAboveBlock => {
+                write!(f, "it cannot suppress formatting for an ambiguous region")
```

Comment:
Is this the terminology we use in the documentation too? `ambigious` region kind of makes sense but is hard to understand.